src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx

```diff
@@ -65,36 +64,43 @@ export function OpenSavedQueryFlyout({
   const [searchQuery, setSearchQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
   const [isLoading, setIsLoading] = useState(false);
   const currentTabIdRef = useRef(selectedTabId);
+  const queryStringManager = getQueryService().queryString;
 
   const fetchAllSavedQueriesForSelectedTab = useCallback(async () => {
     setIsLoading(true);
-    const query = queryStringManager.getQuery();
-    let templateQueries: any[] = [];
+    try {
+      const query = queryStringManager.getQuery();
+      let templateQueries: any[] = [];
 
-    // fetch sample query based on dataset type
-    if (query?.dataset?.type) {
-      templateQueries =
-        (await queryStringManager
-          .getDatasetService()
-          ?.getType(query.dataset.type)
-          ?.getSampleQueries?.()) || [];
+      // fetch sample query based on dataset type
+      if (query?.dataset?.type) {
+        templateQueries =
+          (await queryStringManager
+            .getDatasetService()
+            ?.getType(query.dataset.type)
+            ?.getSampleQueries?.()) || [];
 
-      // Check if any sample query has isTemplate set to true
-      const hasTemplates = templateQueries.some((q) => q?.attributes?.isTemplate);
-      setHasTemplateQueries(hasTemplates);
-    }
+        // Check if any sample query has isTemplate set to true
+        const hasTemplates = templateQueries.some((q) => q?.attributes?.isTemplate);
+        setHasTemplateQueries(hasTemplates);
+      }
 
-    // Set queries based on the current tab
-    if (currentTabIdRef.current === 'mutable-saved-queries') {
-      const allQueries = await savedQueryService.getAllSavedQueries();
-      const mutableSavedQueries = allQueries.filter((q) => !q.attributes.isTemplate);
+      // Set queries based on the current tab
       if (currentTabIdRef.current === 'mutable-saved-queries') {
-        setSavedQueries(mutableSavedQueries);
+        const allQueries = await savedQueryService.getAllSavedQueries();
+        const mutableSavedQueries = allQueries.filter((q) => !q.attributes.isTemplate);
+        if (currentTabIdRef.current === 'mutable-saved-queries') {
```

Comment:
is this required? line 89 should be covering it right?

Reply:
@jowg-amazon could clarify?

Reply:
We were noticing that the getAllSavedQueries call can take a little while to return. If a user switches tabs before this call is finished, we were seeing that `setSavedQueries(mutableSavedQueries)` was overwriting the saved queries incorrectly. This check makes sure that we're still on the mutable saved queries tab.

Reply:
instead of just have a state variable that sets a boolean if it `hasTemplateQueries`, why don't we just have a state variable for templateQueries and a state variable mutableSavedQueries.

Or even more so just set saved queries to all queries and then in the component if the selected tab is template-saved-queries you filter out all the saved queries not matching template-saved-queries etc.